File: src/main/java/macro303/neptunes_pride/Technology.java
package macro303.neptunes_pride;

/**
 * Created by Macro303 on 2018-04-17.
 */
class Technology {
	private double value = 0.0;
	private int level = 0;

	double getValue() {
		return value;
	}

	int getLevel() {
		return level;
	}

	@Override
	public boolean equals(Object o) {
		if (this == o) return true;
		if (!(o instanceof Technology)) return false;

		Technology that = (Technology) o;

		if (Double.compare(that.value, value) != 0) return false;
		return level == that.level;
	}

	@Override
	public int hashCode() {
		int result;
		long temp;
		temp = Double.doubleToLongBits(value);
		result = (int) (temp ^ (temp >>> 32));
		result = 31 * result + level;
		return result;
	}

	@Override
	public String toString() {
		return "Technology{" +
				"value=" + value +
				", level=" + level +
				'}';
	}
}

File: src/main/kotlin/macro303/neptunes_pride/Technology.kt
package macro303.neptunes_pride

/**
 * Created by Macro303 on 2018-04-17.
 */
internal class Technology {
	var value: Double = 0.0
		private set
	var level: Int = 0
		private set

	override fun equals(other: Any?): Boolean {
		if (this === other) return true
		if (other !is Technology) return false

		if (value != other.value) return false
		if (level != other.level) return false

		return true
	}

	override fun hashCode(): Int {
		var result = value.hashCode()
		result = 31 * result + level
		return result
	}

	override fun toString(): String {
		return "Technology(value=$value, level=$level)"
	}
}